### PR TITLE
PAYARA-2470 implement getPropertyNames method of Config correctly

### PR DESCRIPTION
**Pull request by smillidge**

### CODE REVIEW

**Comment by smillidge** [comment]
jenkins test please

**Comment by smillidge** [comment]
jenkins test please

File: appserver/payara-appserver-modules/microprofile/config/src/main/java/fish/payara/microprofile/config/cdi/ConfigPropertyProducer.java [hunk 1]
```diff
@@ -69,7 +71,7 @@ public static final Object getGenericProperty(InjectionPoint ip) {
         Object result = null;
         ConfigProperty property = ip.getAnnotated().getAnnotation(ConfigProperty.class);
         PayaraConfig config = (PayaraConfig) ConfigProvider.getConfig();
-        Class<?> type = (Class<?>) ip.getType();
+        Type type = (Type) ip.getType();
```
Comment (arjantijms) [inline appserver/payara-appserver-modules/microprofile/config/src/main/java/fish/payara/microprofile/config/cdi/ConfigPropertyProducer.java]: Cast to (Type) is not needed here.

**Comment by smillidge** [comment]
jenkins test please

**Pull request merged**